src/btree.c

```diff
@@ -788,6 +790,7 @@ static __le16 init_mid_free_len(int level)
 	free = SCOUTFS_BLOCK_LG_SIZE - sizeof(struct scoutfs_btree_block);
 	if (level == 0)
 		free -= SCOUTFS_BTREE_LEAF_ITEM_HASH_BYTES;
+	free = round_down(free, BTREE_VALUE_ALIGN);
```

Comment:
Let's not round down the calculated hash size, creating unused block bytes.  Let's tweak the _HASH_BYTES define to create sizes that are a multiple of 8 then add a BUILD_BUG_ON in the btree that the hash_bytes is aligned to BTREE_VALUE_ALIGN.